frontend/src/core/codemirror/language/sql.ts

```diff
@@ -131,8 +131,29 @@ export class SQLLanguageAdapter implements LanguageAdapter {
     const showOutputParam = this.showOutput ? "" : ",\n    output=False";
     const engineParam =
       this.engine === this.defaultEngine ? "" : `,\n    engine=${this.engine}`;
-    const end = `\n    """${showOutputParam}${engineParam}\n)`;
 
+    const localTables = store.get(datasetTablesAtom);
+    // Need to check table exists with with boundaries
+    const tablesString = localTables
+      .filter((table) => {
+        const matched = Boolean(
+          new RegExp(`\\b${table.name}\\b`, "g").test(escapedCode),
+        );
+        return table.source_type === "local" && matched;
+      })
+      .map((table) => `"${table.name}": ${table.name}`)
+      .join(",\n        ");
+    // Table params only valid in DuckDB
+    const tablesParam =
+      this.engine === this.defaultEngine && tablesString
+        ? `,\n    tables={\n        ${tablesString}\n    }`
+        : "";
+
+    const end = `\n    """${showOutputParam}${engineParam}${tablesParam}\n)`;
+
+    // TODO: Ruff-wasm is now more main stream (adopted by jupyter-ruff)
```

Comment:
this manual formatting is quite easy (and gets reformatted with black in the backend). im not sure this is a TODO we want to do for the extra dependency

Reply:
Noticed that the line length varies the behavior (ruff rather fit everything on one line if it can). I think the manual formatting may eventually be more effort than it is worth